packages/namers/default/src/DefaultNamer.js

```diff
@@ -1,60 +1,82 @@
-// @flow
+// @flow strict-local
+
+import type {Bundle, BundleGraph, FilePath, NamerOptions} from '@parcel/types';
+
 import {Namer} from '@parcel/plugin';
+import assert from 'assert';
 import crypto from 'crypto';
 import path from 'path';
 
 const COMMON_NAMES = new Set(['index', 'src', 'lib']);
 const DEFAULT_DIST_DIR = 'dist';
 
 export default new Namer({
-  name(bundle, opts) {
+  name(bundle: Bundle, bundleGraph: BundleGraph, opts: NamerOptions): FilePath {
```

Comment:
btw one of the reasons I originally wanted to use `namer({ ... })` or `new Namer({ ... })` was so that type systems could automatically infer these types for you without having to add type annotations to them. 

If you're needing to manually type them, you might want to look into the type definition for `Namer` so that it does the work for you